internal/controller/postgrescluster/apply_test.go

```diff
@@ -79,8 +79,19 @@ func TestServerSideApply(t *testing.T) {
 		after := constructor()
 		assert.NilError(t, cc.Patch(ctx, after, client.Apply, reconciler.Owner))
 		assert.Assert(t, after.GetResourceVersion() != "")
-		assert.Assert(t, after.GetResourceVersion() != before.GetResourceVersion(),
-			"expected https://github.com/kubernetes-sigs/controller-runtime/issues/1356")
+
+		switch {
```

Comment:
:question: Would an if-else work here?

Reply:
Yes, but there is a possibility we'll need additional cases in the future. Also, we use `switch` elsewhere for this type of behavior, so it's more consistent with existing coding patterns.